frontend/src/metabase/entities/containers/rtk-query/EntityObjectLoaderRtkQuery.tsx

```diff
@@ -0,0 +1,289 @@
+import { bindActionCreators } from "@reduxjs/toolkit";
+import type { ComponentType, ReactNode } from "react";
+import { useEffect, useMemo } from "react";
+
+import { skipToken } from "metabase/api";
+import DefaultLoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import {
+  setRequestError,
+  setRequestLoaded,
+  setRequestLoading,
+  setRequestPromise,
+} from "metabase/redux/requests";
+import type { Dispatch } from "metabase-types/store";
+
+import type {
+  EntityDefinition,
+  EntityId,
+  EntityIdSelector,
+  EntityQuery,
+  EntityQuerySelector,
+  EntityType,
+  EntityTypeSelector,
+  FetchType,
+  RequestType,
+} from "./types";
+
+interface ChildrenProps<Entity, EntityWrapper> {
+  // bulkUpdate
+  // create
+  // delete
+  dispatch: Dispatch;
+  dispatchApiErrorEvent: boolean;
+  error: unknown;
+  // fetch
+  // fetchForeignKeys
+  // fetchList
+  // fetchMetadata
+  // fetchMetadataAndForeignTables
+  // fetchMetadataDeprecated
+  fetched: boolean;
+  // invalidateLists
+  loading: boolean;
+  object: EntityWrapper | Entity | undefined; // EntityWrapper when "wrapped" is true, Entity otherwise
+  reload: () => void;
+  // setFieldOrder
+  // table: EntityWrapper;
+  // update
+  // updateProperty
+}
+
+interface LoadingAndErrorWrapperProps {
+  children: ReactNode;
+  loading?: boolean;
+  error?: unknown;
+  noWrapper?: boolean;
+}
+
+export interface Props<Entity, EntityWrapper> {
+  children: (props: ChildrenProps<Entity, EntityWrapper>) => ReactNode;
+  dispatchApiErrorEvent?: boolean;
+  entityAlias?: string;
+  entityId: EntityId | EntityIdSelector | undefined;
+  entityQuery: EntityQuery | EntityQuerySelector;
+  entityType: EntityType | EntityTypeSelector;
+  fetchType?: FetchType;
+  loadingAndErrorWrapper?: boolean;
+  LoadingAndErrorWrapper: ComponentType<LoadingAndErrorWrapperProps>;
+  properties: unknown; // TODO
+  reload?: boolean;
+  // reloadInterval?: (state: State, props: unknown) => number;
+  requestType?: RequestType;
+  selectorName?: string;
+  wrapped?: boolean;
+}
+
+const defaultTransformResponse = (data: unknown, _query: EntityQuery) => data;
+
+/**
+ * Replacement for EntityObjectLoader for better compatibility with RTK Query.
+ *
+ * For example, generic types for the Database entity would be:
+ *   Entity        -> Database from metabase-types/api/database.ts
+ *   EntityWrapper -> Database from metabase-lib/v1/metadata/Database.ts
+ *
+ * @deprecated use "metabase/api" instead
+ */
+export function EntityObjectLoaderRtkQuery<Entity, EntityWrapper>({
```

Comment:
The `RtkQuery` suffix is temporary and is not intended to go to `master`.
The suffix will be removed once the old `EntityObjectLoader` can be removed (#50322).